GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -1289,14 +1395,17 @@ private string GetNewProjectedShaForPlaceholder(string path)
                     metadata.Add("result.RawResult", result.RawResult);
                     metadata.Add("UpdateFailureCause", failureReason.ToString());                    
                     this.context.Tracer.RelatedEvent(EventLevel.Informational, nameof(this.TryRemoveFolderPlaceholder) + "_DeleteFileFailure", metadata);
+
+                    updatedPlaceholderList.TryAdd(placeholder.Path, placeholder);

```

Comment:
I'm glad you mentioned this, the idea was that VFSForGit should still track the placeholder if it fails to delete it, but I think I'll replace this line with a comment that references #245 and update #245 to include handling failures to delete folder placeholders.

I think that on Mac if we fail to delete the placeholder folder we should add its path to ModifiedPaths.dat and show the user a message (otherwise the folder will essentially be abandoned and never be re-expanded or deleted again).

Reply:
(I will dig more into our failure options when working on #245)